client.go

```diff
@@ -0,0 +1,130 @@
+package ndp
+
+import (
+	"fmt"
+	"net"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+)
+
+// A Conn is a Neighbor Discovery Protocol connection.
+type Conn struct {
+	pc *ipv6.PacketConn
+	cm *ipv6.ControlMessage
+
+	allNodes *net.IPAddr
+}
+
+// Dial dials a NDP connection using the specified interface.  It returns
+// a Conn and the link-local IPv6 address of the interface.
+func Dial(ifi *net.Interface) (*Conn, net.IP, error) {
+	llAddr, err := linkLocalAddr(ifi)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ic, err := icmp.ListenPacket("ip6:ipv6-icmp", llAddr.String())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Join the "all nodes" multicast group for this interface.
+	allNodes := &net.IPAddr{
+		IP:   net.IPv6linklocalallnodes,
+		Zone: ifi.Name,
+	}
+
+	pc := ic.IPv6PacketConn()
+	if err := pc.JoinGroup(ifi, allNodes); err != nil {
+		return nil, nil, err
+	}
+
+	// Calculate and place ICMPv6 checksum at correct offset in all messages.
+	if err := pc.SetChecksum(true, 2); err != nil {
+		return nil, nil, err
+	}
+
+	c := &Conn{
+		pc: pc,
+
+		// The default control message used when none is specified.
+		cm: &ipv6.ControlMessage{
+			// Note that hop limit is always 255 in NDP.
+			HopLimit: 255,
```

Comment:
Fascinating, I would have expected it to be 1, to force it to stay link-local. TIL.